### PR TITLE
Upgrade to the latest mirage

### PR DESCRIPTION
**Pull request by jrjohnson**
This is a doozy, most tests had to be modified to work with the new mirage ORM.

Supersedes #2774

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview ready!

Built with commit 7118ce5dd59a412d8cf1888f8635f351f210d0dd

https://deploy-preview-3379--ilios-frontend.netlify.com

File: mirage/factories/curriculum-inventory-sequence.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { Factory } from 'ember-cli-mirage';
+
+export default Factory.extend({
+  description: (i) => `descirption for sequence ${i} `,
```
Comment (stopfstedt) [inline mirage/factories/curriculum-inventory-sequence.js]: typo: should be "description"

File: mirage/helpers/get-all.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
+/*eslint no-console: 0*/
+import getName from './get-name';
+import { Collection, Model } from 'ember-cli-mirage';
+
+export default function getAll(schema, request){
+  //turn /api/programyears?limit=1 into 'programYears'
```
Comment (stopfstedt) [inline mirage/helpers/get-all.js]: please remove comment.

Comment (jrjohnson) [reply mirage/helpers/get-all.js]: This is a comment about the purpose of the regex.

File: mirage/factories/curriculum-inventory-sequence.js (hunk 1 above)
Comment (jrjohnson) [reply mirage/factories/curriculum-inventory-sequence.js]: Existing issue. I will fix post-merge unless there is something else that is going to require running tests again anyway.

**Pull request merged**